### PR TITLE
build: move menu items tree

### PR DESCRIPTION
**Pull request by LinaYahya**
Build move menu items tree to not using mui/ tree view, Basically the user will have a home as a root of the menu which contains own and shared items, 
For every folder that has children within there will be an arrow to let the user navigate to this item.  

image

### CODE REVIEW

File: src/components/main/MoveTreeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+  return (
+    <Dialog
+      onClose={handleClose}
+      aria-labelledby={dialogId}
+      open={open}
+      scroll="paper"
+    >
+      <DialogTitle id={dialogId}>{title}</DialogTitle>
+      <DialogContent sx={{ height: '250px' }}>
```
Comment (pyphilia) [inline src/components/main/MoveTreeModal.tsx:208]: Let's not fix the height. I notice you have the flickering with the width as well anyway.

File: src/components/main/MoveTreeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        <Stack spacing={2} mb={2}>
+          <Breadcrumbs separator="›" aria-label="breadcrumb">
+            {paths.map((ele, index) => (
```
Comment (pyphilia) [inline src/components/main/MoveTreeModal.tsx:211]: Use `parents` here.

File: src/components/main/MoveTreeModal.tsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  defaultParent?: DiscriminatedItem;
+}
+
+const OwnedItemsTree = ({
+  setPaths,
+  setSelectedId,
+  selectedId,
+  defaultSelectedSubItem,
+  itemIds,
+  defaultParent,
```
Comment (pyphilia) [inline src/components/main/MoveTreeModal.tsx:52]: Why can't this be called `parentItem`?

File: src/components/main/MoveTreeModal.tsx [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+  // serious of breadcrumbs
+  const [paths, setPaths] = useState<DiscriminatedItem[]>([]);
+
+  const { data: parentItem } = hooks.useItem(
+    selectedId === TREE_MODAL_MY_ITEMS_ID ? '' : selectedId,
+  );
+  const { data: parents } = hooks.useParents({
+    id: itemIds?.[0],
+  });
+  const [defaultSelectedSubItem, setDefaultSelectedSubItem] =
```
Comment (pyphilia) [inline src/components/main/MoveTreeModal.tsx:187]: In this component, this doesn't need to be called `default`right?

File: src/components/main/MoveTreeModal.tsx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+        <CancelButton onClick={handleClose} />
+        <Button
+          onClick={onClickConfirm}
+          disabled={!parentItem && selectedId !== TREE_MODAL_MY_ITEMS_ID}
+          id={TREE_MODAL_CONFIRM_BUTTON_ID}
+          variant="contained"
+        >
+          {translateBuilder(BUILDER.MOVE_BUTTON)}
+          {(selectedId === TREE_MODAL_MY_ITEMS_ID &&
+            ` to ${translateBuilder(BUILDER.HOME_TITLE)}`) ||
```
Comment (pyphilia) [inline src/components/main/MoveTreeModal.tsx:251]: Let's translate `to` as well.

File: src/components/main/MoveTreeModal.tsx [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+            (parentItem?.name && ` to ${parentItem?.name}`)}
```
Comment (pyphilia) [inline src/components/main/MoveTreeModal.tsx:252]: Same comment as above.

File: src/components/main/RowMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        ?.split('.')
+        .map((ele) => ele.replaceAll('_', '-'));
+
+      const isItemChild = pathsSerious.indexOf(itemIds[i]);
+      if (isItemChild > -1) {
+        return isItemChild > -1;
+      }
+    }
+  }
+  return false;
```
Comment (pyphilia) [inline src/components/main/RowMenu.tsx:44]: You're disabling the option if the folder is a parent right? 
```suggestion
const itemPaths = itemIds.map((ele) => ele.replaceAll('_', '-'));
return itemPaths.some(item => item.path.includes(itemPath))
```
Be careful not to use `?.` everywhere, here you assume `itemPath` exist by typing it.

File: src/components/main/RowMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    return null;
+  }
+  return (
+    <Button
+      onMouseEnter={handleHover}
+      onMouseLeave={handleUnhover}
+      onClick={() => {
+        setSelectedId(ele?.id);
+      }}
+      id={`${TREE_MODAL_MY_ITEMS_ID}-${ele.id}`}
```
Comment (pyphilia) [inline src/components/main/RowMenu.tsx:74]: Factor out in `selectors`.

File: src/components/main/RowMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+      sx={{
+        display: 'flex',
+        color: 'black',
+        width: '100%',
+        justifyContent: 'space-between',
+        marginBottom: '12px',
```
Comment (pyphilia) [inline src/components/main/RowMenu.tsx:80]: Can you use the `theme`?

File: src/components/main/RowMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+        background: selectedId === ele.id ? 'rgba(80, 80, 210, 0.08)' : 'none',
```
Comment (pyphilia) [inline src/components/main/RowMenu.tsx:81]: Factor out the color.

File: src/components/main/RowMenu.tsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      <Box display="flex" gap="4px">
+        <FolderIcon />
+        {ele.name}
+      </Box>
+      {(isHoverActive || selectedId === ele.id) && (
+        <Box display="flex">
+          <Button sx={{ padding: '0' }}>
+            {translateBuilder(BUILDER.MOVE_BUTTON)}
+          </Button>
+          {true && (
```
Comment (pyphilia) [inline src/components/main/RowMenu.tsx:97]: Remove??

File: src/components/main/RowMenu.tsx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+            <IconButton
+              sx={{ padding: '0' }}
+              onClick={(e) => {
+                e.stopPropagation();
```
Comment (pyphilia) [inline src/components/main/RowMenu.tsx:101]: Remove?

File: src/langs/en.json [hunk 13]
```diff
@@ -290,5 +290,6 @@
   "SHORT_LINK_MIN_CHARS_ERROR": "The short link must at least have {{data}} chars.",
   "SHORT_LINK_MAX_CHARS_ERROR": "The short link must not exceed {{data}} chars.",
   "SHORT_LINK_INVALID_CHARS_ERROR": "The short link contains invalid chars ({{data}}).",
-  "SHORT_LINK_UNKNOWN_ERROR": "An unknown error occurred."
+  "SHORT_LINK_UNKNOWN_ERROR": "An unknown error occurred.",
+  "EMPTY_FOLDER_CHILDREN_FOR_THIS_ITEM": "No Sub-Folders For this Item"
```
Comment (pyphilia) [inline src/langs/en.json:294]:
```suggestion
  "EMPTY_FOLDER_CHILDREN_FOR_THIS_ITEM": "This item does not contain any folder."
```

**Review by pyphilia — CHANGES_REQUESTED** [review]
The modal tree is quite complex, I'd like more explanation and I'll review again. Thanks! 🎄

File: src/components/main/MoveTreeModal.tsx [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+  itemIds = [],
+}: TreeModalProps): JSX.Element => {
+  const { t: translateBuilder } = useBuilderTranslation();
+
+  const [selectedId, setSelectedId] = useState<string>('');
+  // serious of breadcrumbs
+  const [paths, setPaths] = useState<DiscriminatedItem[]>([]);
+
+  const { data: parentItem } = hooks.useItem(
+    selectedId === TREE_MODAL_MY_ITEMS_ID ? '' : selectedId,
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/components/main/MoveTreeModal.tsx:195]: `useItem` only receive string or undefined

File: src/components/main/MoveTreeModal.tsx [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+  onClose,
+  onConfirm,
+  open = false,
+  itemIds = [],
+}: TreeModalProps): JSX.Element => {
+  const { t: translateBuilder } = useBuilderTranslation();
+
+  const [selectedId, setSelectedId] = useState<string>('');
+  // serious of breadcrumbs
+  const [paths, setPaths] = useState<DiscriminatedItem[]>([]);
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/components/main/MoveTreeModal.tsx]: I don't prefer to useParents as I have a custom items such as `root` and `home`, and there's a case's that might not connected to the first parent, as an example when we want to move item from nested parent

File: src/components/common/MoveButton.tsx [hunk 16]
```diff
@@ -25,36 +30,78 @@ type MoveButtonProps = {
...
+    // change item's root id to null
+    const newPayload = {
+      ...payload,
+      to:
+        payload.to &&
+        payload.to !== TREE_MODAL_MY_ITEMS_ID &&
+        validate(payload.to)
+          ? payload.to
+          : undefined,
+    };
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/components/common/MoveButton.tsx:67]: this case will not happen as there's always a value for place to move, But this is only as a kind of validation

**Review by pyphilia — CHANGES_REQUESTED** [review]

File: src/components/main/MoveTreeModal.tsx [hunk 17]
```diff
@@ -0,0 +1,149 @@
...
+                  setPaths((prevPaths) => {
+                    const trimmedIndex = prevPaths.indexOf(ele);
+                    if (trimmedIndex === 0) {
+                      return prevPaths.slice(0, -prevPaths.length + 1);
+                    }
+                    return prevPaths.slice(0, trimmedIndex + 1);
+                  });
+
+                  if (ele.id === ROOT_MODAL_ID) {
+                    setSelectedId('');
```
[the remark this answers is not in the document]
Comment (LinaYahya) [reply src/components/main/MoveTreeModal.tsx:120]: Actually the initial value is empty string So I'd prefer to keep it as it's default

**Pull request merged**